python/taichi/tools/sort.py

```diff
@@ -0,0 +1,29 @@
+import taichi as ti
+
+
+def parallel_sort(x):
+    N = x.shape[0]
+
+    @ti.kernel
+    def sort_stage(x: ti.template(), N: int, p: int, k: int, invocations: int):
+        for inv in range(invocations):
+            j = k % p + inv * 2 * k
+            for i in range(0, min(k, N - j - k)):
+                a = i + j
+                b = i + j + k
+                if int(a / (p * 2)) == int(b / (p * 2)):
+                    val_a = x[a]
+                    val_b = x[b]
+                    if val_a > val_b:
+                        x[a] = val_b
+                        x[b] = val_a
+
+    p = 1
+    while p < N:
+        k = p
+        while k >= 1:
+            invocations = int((N - k - k % p) / (2 * k)) + 1
+            sort_stage(x, N, p, k, invocations)
+            ti.sync()
```

Comment:
This simply shouldn't be here, It shouldn't be necessary and will make the program slower. However, on Vulkan, if this is removed, the results are wrong. We probably need to add a `buffer_barrier` or something between consecutive compute dispatches.

Reply:
created #3791 to fix the vulkan problem. After this is fixed, I will submit another PR to remove the `ti.sync()`